### ISSUE TITLE
False positive `nested-min-max`

### ISSUE THREAD

**Issue by XuehaiPan** [PyCQA/pylint#8168]
### Bug description

The `nested-min-max` suggests `max(3, max(1, 2))` to be `max(3, 1, 2)`. However, if the `min` / `max` function takes an iterable `max(iterable)` instead of multiple inputs `max(a, b, c)`. The nested min/max cannot be rewritten as a single statement `max(x, max(iterable)) -> max(x, iterable)`.

```python
"""test.py"""

lst = [1, 2]
max(3, max(lst))
```

If rewrite it to `max(3, lst)`, it will raise:

```pytb
TypeError: '>' not supported between instances of 'list' and 'int'
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:4:0: W3301: Do not use nested call of 'max'; it's possible to do 'max(3, lst)' instead (nested-min-max)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

No warnings.

### Pylint version

```shell
pylint 2.16.0
astroid 2.14.1
Python 3.8.13 (default, Oct 21 2022, 23:50:54)
```

**Comment by clavedeluna** [comment]
I think this makes sense and follows the logic for pylint also not raising the message for these types of inputs:
```
# This is too complicated (for now) as there is no clear better way to write it
max(max(i for i in range(10)), 0)
max(max(max(i for i in range(10)), 0), 1)
```

**Comment by Pierre-Sassoulas** [comment]
What is wrong with ``max(lst + [3])``?

**Comment by nickdrozd** [comment]
> What is wrong with `max(lst + [3])`?

It allocates an extra list object, which might be undesirable.

How about splicing in the list?

```python
lst = [1, 2]
max(3, *lst)
```

**Comment by solney-syn** [comment]
Splatting the list (or allocating a new one) might be ways to avoid the nested min/max - but that's not what the fixer recommends. It says
> it's possible to do 'max(3, lst)'
Which is incorrect in this case, no?

**Comment by nickdrozd** [comment]
Maybe it could suggest `max(3, *lst)` if the value of `lst` could be inferred to be an iterable? Is that feasible?

**Comment by Pierre-Sassoulas** [comment]
@vejkse thank you for reporting. Can you open a new issue please ? It's going to get lost in a closed issue otherwise.
